AppCenterDataStorage/AppCenterDataStorageTests/MSTokenExchangeTests.m

```diff
@@ -124,6 +124,103 @@ - (void)testWhenNoCachedTokenNewTokenIsCached {
   [contextInstanceMock stopMocking];
 }
 
+- (void)testRequestTokenFailedAuthentication {
```

Comment:
The two tests you added are 98% duplicate code. I think there is a good refactor opportunity to make them more readable.

Reply:
Looks `tokenData` is the only difference there so more than 99%. 😄 
However those two are testing different things, I don't think it is necessary to refactor code only for two. It would be better to review entire tests in the file and refactor effectively. Let's don't block it because of this. @dhei What do you think?

Reply:
Okay, I am fine with that.